### PR TITLE
Fix issue with jumping action buttons in WebKit browsers

### LINKED ISSUE #12511: "Create VM" button unclickable (keeps adding spacing, "jumping" down the page)
Cockpit version: 199-1
OS: Ubuntu 19.04 (cockpit installed from Backports)
Page: Virtual Machines

On a freshly installed server, the "Create VM" button, while enabled, is unclickable - the moment the mouse pointer hovers over the button, it "jumps down", adding some empty space to the page. 

Steps to reproduce

1. Install Cockpit 199-1 from disco-backports, with `cockpit-machines`
2. Go to "Virtual Machines" page of cockpit
3. Try to click "Create VM"

Screenshots:

Screenshot 2019-08-07 at 11 05 26
Screenshot 2019-08-07 at 11 05 35
Screenshot 2019-08-07 at 11 05 37
Screenshot 2019-08-07 at 11 05 39
Screenshot 2019-08-07 at 11 05 41

### PR DESCRIPTION
**Pull request by KKoukiou**
Replace <caption> element from cockpit lists because of the following
reasons:
- it's buggy in WebKit
- it skips the a11y tree for headings
- we overload it with stuff that it isn't intended for

Fixes #12511
Related to https://bugs.webkit.org/show_bug.cgi?id=187903.

### CODE REVIEW

**Comment by marusak** [comment]
Thank you for working on this!
This however break layout (you probably know about that, but just in case):
Current master: (seems that on master the `View automation script` is huge, needs fixing)
hugeshit
This PR:
squashed
Current master:
dockerproper
This PR: (notice where is the `Containers` label)
jumpedandspace

**Comment by garrett** [comment]
@marusak: We're using it all over the place, so I expected things would break a bit. I'll see what I can do with the HTML + CSS to address this.

**Comment by garrett** [comment]
@marusak: Please check again. I've changed things with additional commits that fix the layout across all the lists and I reworked SELinux to use a ct-listing HTML pattern. (It could probably be ported to ct-listing, but I did not do that in my commit.)

The only issue that remains now is the Applications page as it does not use ct-listing, but does use `<caption>`. However, that's only cosmetic on load, so it's not critical — but it would be nice to fix as well.

I also need to re-check this all across our different browsers with the latest commits. (However, I have upcoming meetings, so I figured I'd share what I have before the cross-browser checks, so you and @KKoukiou could look through to make sure things are fine and/or improve code further.)

Verified working in:
- [x] Epiphany
- [x] Safari
- [x] Firefox
- [ ] Chrome
- [ ] Edge

**Comment by garrett** [comment]
I've tested the changes across browsers and it works flawless everywhere.

One thing I did see — when there are actions without a heading, the actions are left-aligned. I have a fix I'm testing locally. It's super-simple, but I need to add it to the PR. Expect it soon. :wink:

File: test/verify/check-system-info [hunk 1]
```diff
@@ -412,7 +412,7 @@ class TestSystemInfo(MachineCase):
         else:
             pci_selector = '#hwinfo #pci-listing'
         # PCI
-        b.wait_in_text(pci_selector + ' .listing-ct caption', "PCI")
+        b.wait_in_text(pci_selector + ' .listing-ct-heading', "PCI")
```
Comment (marusak) [inline test/verify/check-system-info]: This breaks [rhel-8-1-distropkg](https://209.132.184.41:8493/logs/pull-12552-20190814-061111-bbf0425d-cockpit-project-cockpit-rhel-8-0-distropkg/log.html#236)

Comment (KKoukiou) [reply test/verify/check-system-info]: Just noticed. Posting a fix, thanks

Comment (KKoukiou) [reply test/verify/check-system-info]: @marusak You confused me, rhel-8-1-distropkg does not exist :P It's the rhel-8-0-distropkg of course. But the point is valid. Thanks!

**Comment by marusak** [comment]
[rhel-8-1](https://209.132.184.41:8493/logs/pull-12552-20190815-061919-5a34b0f7-cockpit-project-cockpit-rhel-8-1/log.html#206) failed [two](https://209.132.184.41:8493/logs/pull-12552-20190815-044444-5a34b0f7-cockpit-project-cockpit-rhel-8-1/log.html#206) times equally on `testCreate (check_machines_dbus.TestMachinesDBus)`, that is kinda suspicious. Once more

**Comment by KKoukiou** [comment]
It's the same libvirt crash that I was trying to catch with https://github.com/cockpit-project/cockpit/commit/cacca69f3f4880f5cd5774864244ec2dcb2c5982

Apparently we need to extend that even more.

See in journal:
Aug 15 03:42:38 localhost.localdomain libvirtd[1644]: 2019-08-15 07:42:38.468+0000: 1646: error : virHashAddOrUpdateEntry:341 : internal error: Duplicate key

**Comment by marusak** [comment]
> Apparently we need to extend that even more

ech, bummer. Can you please take a look at it? :)

**Comment by marusak** [comment]
Currently `testCreate  from check_machines_dbus` is way to flaky everywhere.
The naughty is too specific. One fix is already in #12565 and second one I will send in a bit.
Other test failures are not related.

**Pull request merged**